tensorboard/webapp/plugins/npmi/views/data_selection/metric_search/metric_search_test.ts

```diff
@@ -0,0 +1,257 @@
+      const fixture = TestBed.createComponent(MetricSearchContainer);
+      fixture.detectChanges();
+
+      const input = fixture.debugElement.query(By.css('input'));
+      input.nativeElement.focus();
+      fixture.detectChanges();
+
+      input.nativeElement.value = 'a';
+      input.nativeElement.dispatchEvent(new InputEvent('input', {data: 'a'}));
+      fixture.detectChanges();
+
+      expect(dispatchedActions).toEqual([
+        npmiActions.npmiMetricsRegexChanged({regex: 'a'}),
+      ]);
+    });
+  });
+
+  describe('autocomplete', () => {
+    it('shows all metrics on focus', () => {
+      const fixture = TestBed.createComponent(MetricSearchContainer);
+      fixture.detectChanges();
+
+      const input = fixture.debugElement.query(By.css('input'));
+      input.nativeElement.focus();
+      fixture.detectChanges();
+
+      const optionElements = overlayContainer
+        .getContainerElement()
+        .querySelectorAll('mat-option');
+      const options = Array.from(optionElements).map(
+        (optionEl: Element): DebugElement =>
+          getDebugNode(optionEl) as DebugElement
+      );
+
+      expect(options.map((option) => option.nativeElement.textContent)).toEqual(
+        ['metric_1', 'metric_2']
+      );
+    });
+
+    it('shows remaining metrics if some are already filtered for', () => {
+      store.overrideSelector(getMetricFilters, {
+        metric_1: {max: 1.0, min: -1.0, includeNaN: false},
+      });
+      const fixture = TestBed.createComponent(MetricSearchContainer);
+      fixture.detectChanges();
+
+      const input = fixture.debugElement.query(By.css('input'));
+      input.nativeElement.focus();
+      fixture.detectChanges();
+
+      const optionElements = overlayContainer
+        .getContainerElement()
+        .querySelectorAll('mat-option');
+      const options = Array.from(optionElements).map(
+        (optionEl: Element): DebugElement =>
+          getDebugNode(optionEl) as DebugElement
+      );
+
+      expect(options.map((option) => option.nativeElement.textContent)).toEqual(
+        ['metric_2']
+      );
+    });
+
+    it('renders empty when no metrics match', () => {
+      store.overrideSelector(getMetricsRegex, 'YOU CANNOT MATCH ME');
+      const fixture = TestBed.createComponent(MetricSearchContainer);
+      fixture.detectChanges();
+
+      const input = fixture.debugElement.query(By.css('input'));
+      input.nativeElement.focus();
+      fixture.detectChanges();
+
+      const optionElements = overlayContainer
+        .getContainerElement()
+        .querySelectorAll('mat-option');
+      const options = Array.from(optionElements).map(
+        (optionEl: Element): DebugElement =>
+          getDebugNode(optionEl) as DebugElement
+      );
+
+      expect(options.map((option) => option.nativeElement.textContent)).toEqual(
+        []
+      );
+    });
+
+    it('filters by regex', () => {
+      store.overrideSelector(getMetricsRegex, '[A-Za-z]+_1');
```

Comment:
Since the business logic uses a case insensitive regex (the 'i' flag), should we make a test check that [A-Z] matches [a-z]?

Reply:
Made the test so that this is checked.